webui/src/components/user.jsx

```diff
@@ -156,46 +157,47 @@ export const UserProfile = React.createClass({
 
 
 const TokenList = React.createClass({
-    mixins: [React.addons.PureRenderMixin],
-
     getInitialState() {
         return {
-            name: "",
             token: null,
-            tokenList: [],
         }
     },
 
-    componentWillMount() {
-        serverCache.getUserTokens(tokenList => this.setState({tokenList}));
-    },
-
-    newToken(e) {
-        e.preventDefault();
-        if (!this.state.name) {
+    newToken(tokenName) {
+        if (!tokenName) {
             notifications.danger('Please choose a name for the new token');
         } else {
             notifications.clearAll();
-            serverCache.newUserToken(this.state.name, token => {
-                const tokenList = this.state.tokenList;
-                tokenList.push(token);
-                this.setState({token, tokenList})
+            serverCache.newUserToken(tokenName, token => {
+                notifications.success("A new Token is created");
+                this.setState({token});
             });
         }
     },
 
     renderToken(t) {
-        return <li className="list-group-item" key={t.id}>{t.name}</li>;
+        return <div key={t.id}> <Token token_id={t.id} token_name={t.name} removeToken={this.removeToken} /> </div>
```

Comment:
No wait, although they get a key in 'Token' render method, `key={this.props.token_id}`, each of the elements in the iterator still needs a key in `TokenList`component. That's why I added the div there.